src/components/CollectionSelectList.js

```diff
@@ -172,10 +172,10 @@ class CollectionSelectList extends Component {
             });
         });
         if (collectionToAdd.length > 0) {
-            this.addToCollection(collectionToAdd, collections);
+            this.addToCollection(collectionToAdd[0], collections);
         }
         if (collectionToDeleteFrom.length > 0) {
-            this.deleteFromCollection(collectionToDeleteFrom, collections);
+            this.deleteFromCollection(collectionToDeleteFrom[0], collections);
         }
```

Comment:
What happens if the `collectionToDeleteFrom` array has > 1 entry? Should this iterate and remove all? I'm probably missing some context here.

Reply:
So it is not exactly possible with the way that the select list box works currently.  Though I'm thinking about refactoring that a bit for a different reason (that escapes me now).

Reply:
I experienced the UI today and think I understand the intention better. What happens if someone has a pin in multiple categories, doe sit always remove from the first collection in the list?

Testing this I added two collections, added a pin to both, and the pin is not removed from either collection on refresh.

Reply:
It should be treating any addition or removal as an independent action. But we may have a race condition problem. May need a write lock on that. Unless I'm misunderstanding the issue you ran into?